beetsplug/gmusic.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+# This file is part of beets.
+# Copyright 2017, Tigran Kostandyan.
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+
+"""Uploads files to Google Play Music"""
+
+from __future__ import print_function
+import os.path
+
+from beets.plugins import BeetsPlugin
+from beets import ui
+from beets import config
+from beets.ui import Subcommand
+from gmusicapi import Musicmanager, Mobileclient
+import gmusicapi.clients
+
+
+# Checks for OAuth2 credentials, if they don't exist - performs authorization
+m = Musicmanager()
+if os.path.isfile(gmusicapi.clients.OAUTH_FILEPATH):
+    m.login()
+else:
+    m.perform_oauth()
```

Comment:
Instead of using a global variable for the `Musicmanager`, is it possible to use a field on the `Gmusic` class? That can make debugging much easier.